be/src/exec/olap_scan_node.cpp

```diff
@@ -1361,6 +1368,14 @@ void OlapScanNode::transfer_thread(RuntimeState* state) {
 }
 
 void OlapScanNode::scanner_thread(OlapScanner* scanner) {
+    if (UNLIKELY(_transfer_done)) {
+        _scanner_done = true;
+        std::unique_lock<std::mutex> l(_scan_batches_lock);
+        _running_thread--;
+        _scan_thread_exit_cv.notify_one();
```

Comment:
I think _scan_batch_added_cv should be notified too.

Reply:
Ok,
I thought that `notify_all` was already in `OlapScanNode::close`……but I forgot that `_transfer_done` may also be set to true in `scanner_thread`.